server/models/User.py

```diff
@@ -1,16 +1,52 @@
-from . import db
+from . import db, bcrypt
 from marshmallow import Schema
 
 
 class User(db.Model):
-    __tablename__ = 'customers'
-    name = db.Column(db.String(100), nullable=False)
+    __tablename__ = 'users'
+
     id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(100), nullable=False)
+    email = db.Column(db.String(100), nullable=False, unique=True)
+    password = db.Column(db.String(100), nullable=False)
 
-    def __init__(self, name):
+    def __init__(self, name, email, password):
+        if len(password) < 6:
+            raise ValueError
         self.name = name
+        self.email = email
+        self.password = bcrypt.generate_password_hash(password).decode('UTF-8')
+
+    def __repr__(self):
+        return f"<User #{self.id}: {self.username}, {self.email}>"
+
+    def to_dict(self):
+        return {
+            "id": self.id,
+            "name": self.name,
+            "email": self.email
+        }
+
+    def add_to_database(self):
+        try:
+            db.session.add(self)
+            db.session.commit()
+            return self
+        except IntegrityError:
+            db.session.rollback()
+            return None
+
+    
+    def authenticate(cls, email, password):
```

Comment:
authenticate should have an @classmethod 
When made it into a classMethod I could create new users and login successfully. well done.

Reply:
thanks for finding this. yes, this should be a classmethod